src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -313,7 +313,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			let tokenFillSuccess = await this.fillInOrClearAzureToken(newConnection);
 
 			// If the password is required and still not loaded show the dialog
-			if ((!foundPassword && this._connectionStore.isPasswordRequired(newConnection) && !newConnection.password) || !tokenFillSuccess) {
+			if ((!foundPassword && this._connectionStore.isPasswordRequired(newConnection) && !newConnection.password) && !tokenFillSuccess) {
```

Comment:
Changed this from OR to AND to only show the error dialog when there's no password and no token.

Reply:
Why? If a password is required why would we have an auth token?

And how is this related to the linked issue?

Reply:
My understanding is that this function is attempting to connect to the server, This `if` check is seeing if it has a password and is required. In this case, the server doesn't have a password, but rather has an access token so we should connect with that.

Reply:
So I think the real issue here is that the isPasswordRequired doesn't take into account other auth types for non-MSSQL connections. Look at the check here for isPasswordRequired : https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/connection/common/providerConnectionInfo.ts#L203

Note that we special case MSSQL to only set isPasswordRequired to true if the auth type is SqlAuth - since the other auth types don't require a password.

```
		let isPasswordRequired = optionMetadata.isRequired;
		if (this.providerName === Constants.mssqlProviderName) {
			isPasswordRequired = this.authenticationType === ProviderConnectionInfo.SqlAuthentication && optionMetadata.isRequired;
		}
```

So the actual fix here would be to allow the providers to specify whether a password is required for each auth type instead.